data_diff/cloud/datafold_api.py

```diff
@@ -0,0 +1,238 @@
+    api_key: str
+    host: str = "https://app.datafold.com"
+    timeout: int = 30
+
+    def __post_init__(self):
+        self.host = self.host.rstrip("/")
+        self.headers = {
+            "Authorization": f"Key {self.api_key}",
+            "Content-Type": "application/json",
+        }
+
+    def make_get_request(self, url: str) -> Any:
+        rv = requests.get(url=f"{self.host}/{url}", headers=self.headers, timeout=self.timeout)
+        rv.raise_for_status()
+        return rv
+
+    def make_post_request(self, url: str, payload: Any) -> Any:
+        rv = requests.post(url=f"{self.host}/{url}", headers=self.headers, json=payload, timeout=self.timeout)
+        rv.raise_for_status()
+        return rv
+
+    def get_data_sources(self) -> List[TCloudApiDataSource]:
+        rv = self.make_get_request(url="api/data_sources")
+        rv.raise_for_status()
+        return [TCloudApiDataSource(**item) for item in rv.json()]
+
+    def create_data_source(self, config: TDsConfig) -> TCloudApiDataSource:
+        # TODO: replace an internal url by a public one
+        rv = self.make_post_request(url="api/internal/data_sources", payload=config.dict())
+        return TCloudApiDataSource(**rv.json())
+
+    def get_data_source_schema_config(self) -> List[TCloudApiDataSourceConfigSchema]:
+        # TODO: replace an internal url by a public one
+        rv = self.make_get_request(url="api/internal/data_sources/types")
+        return [
+            TCloudApiDataSourceConfigSchema(
+                name=item["name"],
+                db_type=item["type"],
+                config_schema=TCloudApiDataSourceSchema(
+                    title=item["configuration_schema"]["title"],
+                    properties=item["configuration_schema"]["properties"],
+                    required=item["configuration_schema"]["required"],
+                    secret=item["configuration_schema"]["secret"],
+                ),
+            )
+            for item in rv.json()
+        ]
+
+    def create_data_diff(self, payload: TCloudApiDataDiff) -> int:
+        rv = self.make_post_request(url="api/v1/datadiffs", payload=payload.dict())
+        return rv.json()["id"]
+
+    def poll_data_diff_results(self, diff_id: int) -> TCloudApiDataDiffSummaryResult:
+        summary_results = None
+        start_time = time.time()
+        sleep_interval = 5  # starts at 5 sec
+        max_sleep_interval = 60
+        max_wait_time = 300
+
+        while not summary_results:
+            response = self.make_get_request(url=f"api/v1/datadiffs/{diff_id}/summary_results")
+            response_json = response.json()
+            if response_json["status"] == "success":
+                summary_results = response_json
+            elif response_json["status"] == "failed":
+                raise Exception(f"Diff failed: {str(response_json)}")
+
+            if time.time() - start_time > max_wait_time:
+                raise Exception("Timed out waiting for diff results")
```

Comment:
In both cases here, a user will be confused not knowing what to do next. It would be good to give them a link to the UI and say go investigate the error there (or wait for the results, which can take hours in some cases).